lib/user_agent_parser/parser.rb

```diff
@@ -39,6 +39,36 @@ def parse(user_agent)
       parse_ua(user_agent, os, device)
     end
 
+    def parse_os(user_agent)
+      pattern, match = first_pattern_match(@os_patterns, user_agent)
+
+      if match
+        os_from_pattern_match(pattern, match)
+      else
+        OperatingSystem.new
+      end
+    end
+
+    def parse_device(user_agent)
+      pattern, match = first_pattern_match(@device_patterns, user_agent)
+
+      if match
+        device_from_pattern_match(pattern, match)
+      else
+        Device.new
+      end
+    end
+
+    def parse_ua(user_agent, os = nil, device = nil)
```

Comment:
@daniel-statsig I think the duplicated private method can be deleted. https://github.com/ua-parser/uap-ruby/blob/241085538c3ba165c476d63f48c5886213412088/lib/user_agent_parser/parser.rb#L109-L117

Reply:
Oh sorry, I missed that. Will remove